### PR TITLE
Cli tests

### PR DESCRIPTION
**Pull request by vtnate**
#### Any background context you want to provide?
Is anything really complete until you've written tests for it?
#### What does this PR accomplish?
- Adds tests for cli commands
- Adds the relevant bits of an sdk project_dir for the cli tests to use
- Adds conditional messages to the user whether each command succeeded or failed
#### How should this be manually tested?
`poetry run py.test tests/management/test_uo_des.py`

### CODE REVIEW

File: tests/management/test_uo_des.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        self.scenario_file_path = self.data_dir / 'sdk_project_scraps' / 'baseline_scenario.csv'
+        self.feature_file_path = self.data_dir / 'sdk_project_scraps' / 'example_project.json'
+
+    def test_cli_builds_sys_params(self):
+        # In python 3.8 we can drop the if statement and simplify this to "my_file.unlink(missing_ok=True)"
+        if (self.output_dir / 'test_sys_param.json').exists():
+            (self.output_dir / 'test_sys_param.json').unlink()
+
+        # run subprocess inside the poetry shell to ensure we're using the same venv
+        run(['poetry', 'run', 'uo_des', 'build-sys-param', self.output_dir / 'test_sys_param.json',
```
Comment (macintoshpie) [inline tests/management/test_uo_des.py:20]: I wonder if we could use the click testing instead? Feels a bit strange running poetry commands in the test itself. Or we could import the specific function we're testing and just pass it what we want? Not sure if that'd work.

Either way, we should assert the exit code is `0`

https://click.palletsprojects.com/en/7.x/testing/
```
from click.testing import CliRunner
from management.uo_des import cli

def test_hello_world():
  runner = CliRunner()
  result = runner.invoke(cli, ['build-sys-param', self.output_dir / 'test_sys_param.json'])
```

File: management/uo_des.py [hunk 2]
```diff
@@ -235,6 +240,11 @@ def create_model(model_type, sys_param_file, geojson_feature_file, project_name)
     )
     district.to_modelica()
 
+    if (Path(project_name) / 'Districts' / 'DistrictEnergySystem.mo').exists():
```
Comment (macintoshpie) [inline management/uo_des.py:243]: Same question as above, wouldn't it have raised an exception before here or have run successfully?

**Review by macintoshpie — COMMENTED** [review]
This is looking good, I just a few questions. Also, I'm wondering if all of these files being added under `sdk_project_scraps` are really necessary? Are we using all of them when testing?

File: management/uo_des.py [hunk 3]
```diff
@@ -264,3 +274,8 @@ def run_model(modelica_project):
     # setup modelica runner
     mr = ModelicaRunner()
     mr.run_in_docker(file_to_run, run_path=run_path.parent, project_name=project_name)
+
+    if (run_path.parent / f'{project_name}_results' / f'{project_name}_Districts_DistrictEnergySystem_result.mat').exists():
+        print(f"\nModelica model {project_name} ran successfully")
+    else:
+        raise SystemExit(f"{project_name} had a problem. Please check your inputs and try again.")
```
[the remark this answers is not in the document]
Comment (vtnate) [reply management/uo_des.py:281]: That's a good philosophical question. I guess my test is more like an integration test whereas you're suggesting a unit test. Is there benefit to having both?

[the remark this answers is not in the document]
Comment (macintoshpie) [reply management/uo_des.py:281]: Either way is probably fine here 🤷‍♂️

File: management/uo_des.py [hunk 4]
```diff
@@ -131,6 +131,11 @@ def build_sys_param(model_type, sys_param_filename, scenario_file, feature_file,
         overwrite=overwrite
     )
 
+    if Path(sys_param_filename).exists():
```
[the remark this answers is not in the document]
Comment (vtnate) [reply management/uo_des.py:134]: You're right, it does do one of those things. This check is for providing explanation to the user.

File: management/uo_des.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vtnate) [reply management/uo_des.py:281]: I'm going to keep these tests basically as is to catch integration issues. Thanks for thinking about it!

File: tests/management/test_uo_des.py (hunk 1 above)
Comment (vtnate) [reply tests/management/test_uo_des.py:20]: Good point about including poetry in the test. I removed that bit from the subprocess.run command. I couldn't get the click.testing bit working quickly, so I'm going to table that for now.

File: management/uo_des.py (hunk 2 above)
Comment (vtnate) [reply management/uo_des.py:243]: Yep, I'm using the check to provide a helpful message to the user.

**Comment by vtnate** [comment]
@macintoshpie I made a couple changes from your suggestions, and I'd like to keep the tests more like integration rather than unit. Is it ok as it is now?

**Review by macintoshpie — APPROVED** [review]
Nice 😎

**Pull request merged**